### PR TITLE
Add public getter for IndexMetadata::identifier [ECR-4163]

### PR DESCRIPTION
**Pull request by vitvakatu**
## Overview

I decided to move documentation to the method, so it would be available on docs.rs

---

See: https://jira.bf.local/browse/ECR-4163

### Definition of Done

- [ ] There are no TODOs left in the merged code
- [ ] Change is covered by automated tests
- [ ] Benchmark results are attached (if applicable)
- [ ] The [coding guidelines] are followed
- [ ] Public API has proper documentation
- [ ] Changelog is updated if needed (in case of notable or breaking changes)
- [ ] The continuous integration build passes

[coding guidelines]: https://github.com/exonum/exonum/blob/master/CONTRIBUTING.md#conventions

### CODE REVIEW

File: components/merkledb/src/views/metadata.rs [hunk 1]
```diff
@@ -241,6 +237,16 @@ impl<V> IndexMetadata<V> {
     pub fn index_type(&self) -> IndexType {
         self.index_type
     }
+
+    /// Returns a globally unique numeric index identifier.
+    /// MerkleDB assigns a unique numeric ID for each fully-qualified index name.
+    ///
+    /// Valid identifiers are non-zero.
```
Comment (dmitry-timofeev) [inline components/merkledb/src/views/metadata.rs]: Shall we enforce it (in the constructor of this structure)? Or move the last two sentences to the thing that enforces these properties and link from here (@slowli , @popzxc)?

Comment (popzxc) [reply components/merkledb/src/views/metadata.rs]: I'd prefer the latter option.

Comment (vitvakatu) [reply components/merkledb/src/views/metadata.rs]: `IndexesPool::create_index_metadata` has a comment saying "Identifier should be non-zero to translate to a correct id in `ResolvedAddress`". `ResolveAddress::new` has a corresponding signature `(name, id: Option<NonZeroU64>)`. I'm not sure where should I put these sentences.

Comment (slowli) [reply components/merkledb/src/views/metadata.rs]: A provocative idea: Is it possible to change identifier to `NonZeroU64`?

Comment (vitvakatu) [reply components/merkledb/src/views/metadata.rs]: So the `1` would be the initial value?

Comment (slowli) [reply components/merkledb/src/views/metadata.rs]: I mean define `IndexMetadata.identifier` as `NonZeroU64` and return `NonZeroU64` from the corresponding public method.

File: components/merkledb/src/views/metadata.rs [hunk 2]
```diff
@@ -340,7 +346,7 @@ impl<T: RawAccess> IndexesPool<T> {
     {
         let len = self.len();
         let metadata = IndexMetadata {
-            identifier: len + 1,
+            identifier: NonZeroU64::new(len + 1).unwrap(),
             // ^-- Identifier should be non-zero to translate to a correct id in `ResolvedAddress`
```
Comment (slowli) [inline components/merkledb/src/views/metadata.rs]: Nit / opinion: I feel that the comment is obsolete now since `NonZeroU64::new` is self-explanatory.

**Pull request merged**